### PR TITLE
feat(connection-limit): add function to mutate `ConnectionLimits`

### LINKED ISSUE #4826: connection-limits: Add functions to update `ConnectionLimits`
### Description

Add functions in `libp2p-connection-limits` to update `ConnectionLimits`. 

### Motivation

This would allow developers to update the limits accordingly without needing to restart the node to apply the changes. 

### Current Implementation

`libp2p-connection-limits` has limits that are fixed and can only be adjusted/modified prior to passing it to the behaviour. 

### Are you planning to do it yourself in a pull request ?

Yes

### PR DESCRIPTION
**Pull request by dariusc93**
## Description

Resolves: #4826.

## Notes & open questions

Should we note that this will not close any connections that exceeds the connection limit if the limit is lowered? 

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: misc/connection-limits/src/lib.rs [hunk 1]
```diff
@@ -80,6 +80,11 @@ impl Behaviour {
         }
     }
 
+    /// Returns a mutable reference to [`ConnectionLimits`].
```
Comment (thomaseizinger) [inline misc/connection-limits/src/lib.rs:83]: Let's add some docs here that explain that the new limits are not directly enforced.

File: misc/connection-limits/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## 0.3.1 - unreleased
```
Comment (thomaseizinger) [inline misc/connection-limits/CHANGELOG.md]:
```suggestion
## 0.3.1
```

See https://github.com/libp2p/rust-libp2p/pull/4894.

**Review by thomaseizinger — COMMENTED** [review]
Thanks! Just two minor requests.

File: Cargo.toml [hunk 3]
```diff
@@ -75,8 +75,8 @@ futures-bounded = { version = "0.2.3", path = "misc/futures-bounded" }
 libp2p = { version = "0.53.2", path = "libp2p" }
 libp2p-allow-block-list = { version = "0.3.0", path = "misc/allow-block-list" }
 libp2p-autonat = { version = "0.12.0", path = "protocols/autonat" }
-libp2p-connection-limits = { version = "0.3.0", path = "misc/connection-limits" }
-libp2p-core = { version = "0.41.2", path = "core" }
+libp2p-connection-limits = { version = "0.3.1", path = "misc/connection-limits" }
+libp2p-core = { version = "0.41.1", path = "core" }
```
Comment (thomaseizinger) [inline Cargo.toml]:
```suggestion
libp2p-core = { version = "0.41.2", path = "core" }
```

That doesn't seem correct!

Comment (dariusc93) [reply Cargo.toml]: Thanks. Mustve happened during one of the merges

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**